### ISSUE TITLE
Address changes after Rails 5.0

### ISSUE THREAD

**Issue by Throne3d** [glowfic-constellation/glowfic#476]
After #448 there are some things still to be addressed – this is a continuation issue.

Deprecation:
- [x] `ActiveRecord::Base.raise_in_transactional_callbacks=` (set to true, now forced true and the config does nothing; removed)
- [x] Parameters are now objects instead of hashes and comparing equality is deprecated set to be removed in Rails 5.1
- [x] `alias_method_chain` deprecated (in Presentable)
- [x] `before_filter`, `around_filter`, `after_filter` in basically all the controllers
- [x] Positional parameters in tests (`get :show, id: 1` vs `get :show, params: { id: 1 }`)
- [x] Rails target versions in migration superclasses (`< ActiveRecord::Migration[4.2]`)
- [x] `uniq!` → `distinct!` (`app/controllers/messages_controller.rb:14`)

Later:
- [x] Fix API tests to use `as: :json` and `response.parsed_body` instead of monkey patching `ActiveDispatch::TestResponse#json` - #1847
- [x] Investigate ActiveJob, convert jobs to it if relevant [#475]
- [x] Make `ApplicationJob` for ActiveJob to inherit from ([link](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#activejob-now-inherits-from-applicationjob-by-default)) [#475]
- [x] Comb over `before_…` callbacks for returning `false` – this should now `throw(:abort)` instead
- [x] Handle the upcoming change where `belongs_to` requires an association by default gracefully ([link](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#active-record-belongs-to-required-by-default-option))
- [x] Investigate output options for `db:schema:dump`? ([link](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#configure-the-output-of-db-structure-dump))
- [x] Remove manual dependency on `redis ~> 3.0` (#507)
- [ ] Move to request specs per #57 (#728)
- [x] Have Rails serialize the models for jobs? (It can do that now; we don't need to pass `.id` and get it to retrieve the model.)
- [x] Use the `async` queue adapter in development instead of Resque? (`heroku local` runs Resque anyway, so it might not be necessary, but with `async` presumably it also gets run on `rails s`.)

**Comment by Throne3d** [comment]
Now has remaining tasks from #475

**Comment by Throne3d** [comment]
This mostly doesn't need to be a single issue any more. It should be broken down into its components or attached into other issues.

**Comment by Throne3d** [comment]
All steps done except request specs (separate issue), closing
